ci: Move tidy to persistent worker

maflcko opened the pull request:
Cirrus CI will be capping the free compute soon. For now, switch more tasks to persistent worker, as recommended by Cirrus CI.

(See slightly related discussion in https://github.com/bitcoin/bitcoin/issues/28098)

Also, add more docs.

Commit:
ci: Move tidy to persistent worker

hebasto commented:
macOS task has not been started:
![image](https://github.com/bitcoin/bitcoin/assets/32963518/b92441c3-3864-4e52-895e-84e1de39e983)

hebasto reviewed (approved):
ACK fa960fefb589ac97de206f46d7b5c565fc70e3db. I have reviewed the code and it looks OK.

However, I'm a bit reluctant about running macOS-cross job on a persistent worker. I'd prefer run it on GitHub Actions due to the ability to create downloadable artifacts that is used in the QML repo to make designers' life a bit easier.

I understand, that there is no open PR with such a suggestion for now. But there is an ongoing Linux jobs related discussion in https://github.com/bitcoin-core/secp256k1/pull/1396.

It would be nice if we postpone moving macOS-cross job for a week or two.

hebasto reviewed (approved):
re-ACK fa1d8955f69d3934f975f42eb04b5a3fc0d8aa35.

Thank you.

fanquake commented:
>  I'd prefer run it on GitHub Actions due to the ability to create downloadable artifacts that is used in the QML repo to make designers' life a bit easier.

Why can't this just be done in the QML repo if it's required there?

pinheadmz reviewed (approved):
ACK fa1d8955f69d3934f975f42eb04b5a3fc0d8aa35

Lots of great comments, code change looks good to me and obviously works on ci. I also ran the task locally (although on macOS I had to enter the container and manually install clang with arm support).
One question below and also I'm wondering if the output of "include-what-you-use" is used at all? It makes a lot of suggestions but none of that counts as errors?

<details><summary>Show Signature</summary>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

ACK fa1d8955f69d3934f975f42eb04b5a3fc0d8aa35
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE5hdzzW4BBA4vG9eM5+KYS2KJyToFAmTT4bgACgkQ5+KYS2KJ
yTqpug//S8FFIisKLeNiyUR8DjdLK+twfLOvkRu8iVwBsqgTXFQXxWY6xzbDkqTB
xGpyIGwpJWrREnu9ec2Ru4R8+PPcLwGmKiSET12XfUHWvinrZe0O0nUl1QlckV/h
shmsJkUzF9Zw839mZXL7UevQCg2QfMXYw+iMYgc0dGXL0qncsWAhphfr6LDCaf+h
zNEwhx/KfiPszQPlpeb3lJvV4joE4vGJEvUZ2FAt5vRB89b+68tjL4D3l+yKFhiD
bVT9k7D+nAAmm2suuXM6CSyQBZPEEknhrgK3OfLzDS/RZdrBoDSkrae4KsM9nENs
1TJ0K/x9tZCjQlmTKjEEdOgxv1kCmcQkSwqq57k5q5BjbyfVCAxxYhN7LB0kUdnr
Ynjs/wzAVYb+kPsWvBi0gbEsl4LuFEGSMsSFTsowtoDlA49TNjTTQwzJMbky9nPh
6A5LBUAZRPlD6rDorEUoxL85Amj/DiL52xsHKF9qnc+yI0n26/xv8gm8WTD7tKc0
jI2GYVSCD23VBO0wYvan8cxbBDLAjF13eEVtFT4r31RBTxoyWiG0Ty+qd3qMhvKE
JspOA2sAscWFIjlFVH0AOyvMDU4o1rU7XLHCvFGc9Gj9imWxIpHEOTy0A8q78oSn
tW4czoaWWXmC3tHphs2/qA90txKM8lUuqi8sl+aqhUqSo8lv1Eg=
=If5B
-----END PGP SIGNATURE-----
```

pinheadmz's public key is [on keybase](https://keybase.io/pinheadmz/pgp_keys.asc?fingerprint=e61773cd6e01040e2f1bd78ce7e2984b6289c93a)
</details>

hebasto commented:
> I'll create a separate pull/thread for macOS-cross next week, so that it can be discussed.

For a proposal to move macOS-cross to GitHub Actions, please see https://github.com/bitcoin/bitcoin/pull/28265.

hebasto commented:
> > One question below and also I'm wondering if the output of "include-what-you-use" is used at all? It makes a lot of suggestions but none of that counts as errors?
> 
> Yeah, that is on purpose and unchanged in this pull. Currently iwyu still has bugs (_false_ negatives and _false_ positives) and there are also trillion _true_ positives and the _true_ negatives sometimes lead to compile errors down the line in other modules.
> 
> I think for now it is best to check the iwyu output during review for newly added code or when a module is modified and touched for other reasons.

Related:
- https://github.com/bitcoin/bitcoin/pull/26763
- https://github.com/bitcoin/bitcoin/pull/27710

Commit:
refactor: Remove PERSISTENT_WORKER_* yaml templates

* PERSISTENT_WORKER_TEMPLATE_ENV is not needed at all, because
  RESTART_CI_DOCKER_BEFORE_RUN is already set on the persistent worker.
* PERSISTENT_WORKER_TEMPLATE can be replaced by pinning the
  previous_releases task to a type of worker. This should make the CI
  performance more consistent.

hebasto reviewed (approved):
re-ACK faaa0794b24f250f787bc9b9605270108ea101a0
